src/jbmc/jbmc_parse_options.cpp

```diff
@@ -606,9 +606,14 @@ int jbmc_parse_optionst::get_goto_program(
     }
 
     // show it?
-    if(cmdline.isset("show-goto-functions"))
+    if(
+      cmdline.isset("show-goto-functions") ||
+      cmdline.isset("list-goto-functions"))
     {
-      show_goto_functions(*goto_model, ui_message_handler.get_ui());
+      show_goto_functions(
+        goto_model,
+        ui_message_handler.get_ui(),
+        cmdline.isset("list-goto-functions"));
       return 0;
     }
```

Comment:
May I suggest
````c++
bool show_goto_functions_cmd(
  const cmdlinet &cmdline,
  const goto_modelt &goto_model,
  const ui_message_handlert::uit ui,
  message_handlert &message_handler)
{
  const bool list_gf=cmdline.isset("list-goto-functions");
  if(!cmdline.isset("show-goto-functions") && !list_gf)
    return false;

  show_goto_functions(goto_model, ui, message_handler, list_gf);

  return true;
}
````

and replace the code in each front-end (except for goto-diff) by

````c++
if(
  show_goto_functions_cmd(
    cmdline,
    goto_model,
    ui_message_handler.get_ui(),
    get_message_handler()))
  return CPROVER_EXIT_SUCCESS;
````